homeassistant/components/webdav/camera.py

```diff
@@ -0,0 +1,242 @@
+"""
+This component models a WebDAV share full of images as a camera.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.webdav/
+"""
+
+import logging
+
+import asyncio
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.components.camera import (Camera,
+                                             PLATFORM_SCHEMA, SUPPORT_ON_OFF,
+                                             SUPPORT_STREAM)
+from homeassistant.const import (
+    CONF_HOST, CONF_USERNAME, CONF_PASSWORD, CONF_PATH, CONF_NAME, CONF_TOKEN
+)
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['webdavclient3==0.10']
+
+CONF_CERTIFICATE_PATH = "ssl_client_certificate"
+CONF_IMAGE_INTERVAL = "image_interval"
+CONF_KEY_PATH = "ssl_client_key"
+
+IMAGE_URL = '/api/camera_proxy/{0}?token={1}&frame={2}'
+STREAM_URL = '/api/camera_proxy_stream/{0}?token={1}'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_USERNAME): cv.string,
+    vol.Optional(CONF_PASSWORD): cv.string,
+    vol.Optional(CONF_TOKEN): cv.string,
+    vol.Optional(CONF_CERTIFICATE_PATH): cv.string,
+    vol.Optional(CONF_KEY_PATH): cv.string,
+    vol.Optional(CONF_PATH, default="/"): cv.string,
+    vol.Required(CONF_IMAGE_INTERVAL): cv.time_period,
+})
+
+# This must be separate from frame interval since listing can be expensive
+SCAN_INTERVAL = timedelta(minutes=15)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a single webdav camera."""
+    from webdav3.client import Client
+    from webdav3.exceptions import WebDavException
+
+    name = config[CONF_NAME]
+    host = config[CONF_HOST]
+    path = config.get(CONF_PATH)
```

Comment:
Use [] for accessing required or default keys.

Reply:
Will fix.